src/udev/udev-builtin-net_id.c

```diff
@@ -373,20 +394,26 @@ static int dev_pci_slot(sd_device *dev, const LinkInfo *info, NetNames *names) {
         if (l == 0)
                 names->pci_path[0] = '\0';
 
+        log_device_debug(dev, "PCI path identifier: domain=%u bus=%u slot=%u func=%u phys_port=%s dev_port=%lu → %s",
+                         domain, bus, slot, func, strempty(info->phys_port_name), dev_port,
+                         empty_to_na(names->pci_path));
+
         /* ACPI _SUN — slot user number */
         r = sd_device_new_from_subsystem_sysname(&pci, "subsystem", "pci");
         if (r < 0)
-                return r;
+                return log_debug_errno(r, "sd_device_new_from_subsystem_sysname failed: %m");
 
         r = sd_device_get_syspath(pci, &syspath);
         if (r < 0)
-                return r;
+                return log_device_debug_errno(pci, r, "sd_device_get_syspath failed: %m");
+
         if (!snprintf_ok(slots, sizeof slots, "%s/slots", syspath))
-                return -ENAMETOOLONG;
+                return log_device_debug_errno(dev, SYNTHETIC_ERRNO(ENAMETOOLONG),
+                                              "Cannot access %s/slots: %m", syspath);
 
         dir = opendir(slots);
         if (!dir)
-                return -errno;
+                return log_device_debug_errno(dev, errno, "Cannot access %s: %m", slots);
```

Comment:
Also here.

Reply:
Changed.

Reply:
Actually no, I think it's better to use `dev`. The path we couldn't opened is printed, so this should be unambiguous.